### PR TITLE
feat: Register @route classes for Spring native automatically

### PR DESCRIPTION
**Pull request by Artur-**

### CODE REVIEW

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/springnative/VaadinBeanFactoryInitializationAotProcessor.java [hunk 1]
```diff
@@ -31,9 +36,25 @@
...
+    }
+
     @Override
     public BeanFactoryInitializationAotContribution processAheadOfTime(
             ConfigurableListableBeanFactory beanFactory) {
+        // Find and register @Route classes so they can be created as beans at
+        // runtime
+        if (beanFactory instanceof BeanDefinitionRegistry) {
+            findAndRegisterRoutes(
+                    (BeanDefinitionRegistry & BeanFactory) beanFactory);
```
Comment (mshabarov) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/springnative/VaadinBeanFactoryInitializationAotProcessor.java:50]: Why `BeanFactory` is mentioned here? `ConfigurableListableBeanFactory` extends `BeanFactory` so no explicit cast is needed.

Comment (mshabarov) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/springnative/VaadinBeanFactoryInitializationAotProcessor.java:50]: Ah, so it used later on as a constraint for a generic type. Never mind.

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/springnative/VaadinBeanFactoryInitializationAotProcessor.java [hunk 2]
```diff
@@ -78,6 +99,60 @@ public BeanFactoryInitializationAotContribution processAheadOfTime(
...
+
+        for (String pkg : getPackagesWithRoutes(beanFactory)) {
+            logger.debug("Scanning for @{} or @{} annotated beans in {}",
+                    Route.class.getSimpleName(),
+                    RouteAlias.class.getSimpleName(), pkg);
+            var reflections = new Reflections(pkg);
+            for (var c : getRouteTypesFor(reflections, pkg)) {
+                if (registeredClasses.contains(c.getName())) {
+                    logger.debug(
+                            "Skipping route class {} as it has already been registered as a bean",
```
Comment (mshabarov) [inline vaadin-spring/src/main/java/com/vaadin/flow/spring/springnative/VaadinBeanFactoryInitializationAotProcessor.java:137]: Just for clarification: in which cases we might have the routes not registered as a beans in the spring based application?

**Review by mshabarov — APPROVED** [review]

File: vaadin-spring/src/main/java/com/vaadin/flow/spring/springnative/VaadinBeanFactoryInitializationAotProcessor.java (hunk 2 above)
Comment (Artur-) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/springnative/VaadinBeanFactoryInitializationAotProcessor.java:137]: If you annotate the route class with `@Component` yourself

Comment (mshabarov) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/springnative/VaadinBeanFactoryInitializationAotProcessor.java:137]: How might be that the `@Component` doesn't register a bean? That's what the annotation is for, or we talk about different things?

Comment (Artur-) [reply vaadin-spring/src/main/java/com/vaadin/flow/spring/springnative/VaadinBeanFactoryInitializationAotProcessor.java:137]: Ah sorry, you asked the other way around. In the normal case routes are not registered as beans. They are just created using Spring

**Pull request merged**